flink/v1.18/flink/src/main/java/org/apache/iceberg/flink/source/StreamingMonitorFunction.java

```diff
@@ -195,7 +192,10 @@ void monitorAndForwardSplits() {
     // Refresh the table to get the latest committed snapshot.
     table.refresh();
 
-    Snapshot snapshot = table.currentSnapshot();
+    Snapshot snapshot =
```

Comment:
I am wondering if we should change the default to `SnapshotRef.MAIN_BRANCH`? This way we can simplify the code here a bit.

```
  public static final ConfigOption<String> BRANCH =
      ConfigOptions.key("branch").stringType().defaultValue(null);
```

Reply:
I also just found out `FlinkWriteOptions` has the default to main branch already. so we probably should make it consistent in `FlinkReadOptions`

Reply:
sounds good

Reply:
Since using `SnapshotRef.MAIN_BRANCH` comes with other issues (namely metadata table queries start to fail),  I suggest to merges @rodmeneses's original version. If we want to use `SnapshotRef.MAIN_BRANCH` as a default we can do it in another PR